### ISSUE TITLE
Focus Navigation Slow,Delayed & Unexpected Behaviour

### ISSUE THREAD

**Issue by liammccann** [blackberry/WebWorks#120]
Hi Support,

Posting this regarding help with an application we are building on Webworks for Smartphone’s. The aim of the application is just simple form usage with navigation focus mode enabled allowing the user to navigate around our application as if it was a native application like Blackberry Emails without having to use a cursor.

Our setup information is listed below.
Webworks Version: WebWorks SDK 2.3.1.5

We have tried this on devices including
·         Blackberry Bold 9780
6.0 Budle 2475( v6.0.0.546, Platform 6.6.0.207)
·         Blackberry Bold 9900
7.1 Bundle 1149(v7.1.0.342, Platform 5.1.0.276)
·         Blackberry 9790
7.0 Bundle 2356 (v7.0.0.557, Platform 9.16.0.67)

It has also been tested on another two OS7 devices (9900,9360)  to find the same results as listed below.

When we are building our application we run the following command....
 bbwp Archivename.zip -g password -o ~/"

If you using focus navigation are there anything extra you must include in the command line build?

The issues that seem to be occurring randomly are…
Delay on entering text
If you don’t wait 1 or 2 seconds it wont allow you to enter text in the input text box at all unless you move to another input and then back
When text has already been entered to that field and you scroll back to it sometimes it doesn’t allow you to make changes e.g. add more text or delete text
Overall slow performance when navigation focus mode is enabled, if we disable the application runs the way it should
Our sourcecode looks like...

You think such a basic and important feature would work without any issues.

Our test details,sourcecode,complied applciations are here : https://www.dropbox.com/sh/z1zmezi9j0ihw9u/6YjoDkVhSG 

Thanks for your support,
Liam McCann

**Comment by tneil** [comment]
It appears as though the input box is not gaining focus.  I've noticed that if you click the trackpad before entering your text there will be no delay.

Tried this with both BB7 Torch and BB6 Bold 9780

**Comment by liammccann** [comment]
When clicking before we enter each field it seems to work fine most of the time but still shows unexpected behaviour randomly and doesn't allow you to enter text sometimes.

For example..
If you click one , enter text fine
Then track-pad down one and forget to click it doesn't allow you to enter text

If its any help to you, the navmode.js that you made seemed to solve most of this issue but cost other bugs such as button clicking only worked sometimes. Overall focus based navigation seems to be very buggy as cursor based navigation works perfect with/without bbUI.

Without or without bbUI this seems to be a problem, it would be great to be able to have focus based navigation and bbUI side by side as that would create a nearly native application feel.

Thanks all the help so far,

**Comment by liammccann** [comment]
Has there been any updates on this issue yet? Are RIM working on a fix or is the a trail solution yet?

Thanks again

**Comment by liammccann** [comment]
Any updates yet? This is a key feature and has been over a month now...

**Comment by tneil** [comment]
I don't believe anyone has looked into this yet.. all engineers are currently focused on the BB10 WebWorks coding
